androidsdk/src/main/java/com/wootric/androidsdk/network/tasks/GetEndUserByEmailTask.java

```diff
@@ -37,6 +37,7 @@ protected void onSuccess(String response) {
         if(response == null) {
             RuntimeException responseEmpty = new IllegalArgumentException("End user params are missing");
             wootricApiCallback.onApiError(responseEmpty);
+            return;
```

Comment:
I see what you're going for with the early return to avoid response being null. I'd probably refactor a little bit like this so the error case is always handled in the else block.

```java
protected void onSuccess(String response) {
  if (response != null) {
    try {
      JSONArray jsonArray = new JSONArray(response);

      if (jsonArray.length() > 0) {
        JSONObject jsonObject = jsonArray.getJSONObject(0);
        long endUserId = jsonObject.getLong("id");
        wootricApiCallback.onGetEndUserIdSuccess(endUserId);
      } else {
        wootricApiCallback.onEndUserNotFound();
      }
    } catch (JSONException e) {
      wootricApiCallback.onApiError(e);
    }
  } else {
    wootricApiCallback.onApiError(
      new IllegalArgumentException("End user params are missing"));
  }
}
```